template_handler/cdk.py

```diff
@@ -0,0 +1,199 @@
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+        )
+
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        if credentials.token is None:
+            envs.pop('AWS_SESSION_TOKEN', None)
+        else:
+            envs['AWS_SESSION_TOKEN'] = credentials.token
+
+        return envs
+
+    def _cmd_exists(self, cmd) -> bool:
+        """
+        Checks whether a specified CLI command exists
+        """
+
+        cmd_exists = shutil.which(cmd) is not None
+        self.logger.debug(f"command '{cmd}' exists: {cmd_exists}")
+        return cmd_exists
+
+    def _node_package_exists(self, package) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler
+        """
+
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
+        ]
+        for cmd_prerequisite in cmd_prerequisites:
+            if not self._cmd_exists(cmd_prerequisite):
+                raise exceptions.SceptreException(f"Command prerequisite '{cmd_prerequisite}' not found")
+
+        # Check Node Package Prerequisites
+        node_prerequisites = [
+            'cdk-assets'
+        ]
+        for node_prerequisite in node_prerequisites:
+            if not self._node_package_exists(node_prerequisite):
+                raise exceptions.SceptreException(f"Node Package prerequisite '{node_prerequisite}' not found")
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns
+        -------
+        str|bytes
+            CDK synthesised CloudFormation template
+        """
+
+        self._check_prerequisites()
+
+        # Import CDK Python template module
+        template_path = posixpath.join('templates', self.cdk_template_path)
+        self.logger.debug(f'Importing CDK Python template module {template_path}')
+        template_module_name = 'cdk_template'
```

Comment:
I'm unfamiliar with using importlib, but I'm wondering what this means here? What if there are _multiple_ cdk stacks defined by the user? Since Python inherently caches imports in a global module cache, I'm wondering if one template handler would step on another template handler if they both had identical module names. Why not use the filename from the path instead?

Reply:
I've tested with multiple CDK stacks in a single `sceptre launch` and they deploy fine. I've changed this to use the file name anyway.